tests/kernel/threads/thread_apis/src/test_essential_thread.c

```diff
@@ -47,3 +48,48 @@ void test_essential_thread_operation(void)
 	k_sem_take(&sync_sem, K_FOREVER);
 	k_thread_abort(tid);
 }
+
+void k_sys_fatal_error_handler(unsigned int reason,
+				      const z_arch_esf_t *esf)
+{
+	ARG_UNUSED(esf);
+	ARG_UNUSED(reason);
+
+	z_thread_essential_clear();
+}
+
+static void abort_thread_entry(void *p1, void *p2, void *p3)
+{
+	z_thread_essential_set();
+
+	if (z_is_thread_essential()) {
+		k_busy_wait(100);
+	} else {
+		zassert_unreachable("The thread is not set as essential");
+	}
+
+	k_sem_give(&sync_sem);
+}
```

Comment:
You want to sleep or otherwise block here.  If this thread is allowed to run to completion, then the Zephyr exit code will disable the essential flag before self-aborting, and the abort call below might succeed where you expect it to fail.  You need to be 100% sure that the thread is still running when you call k_thread_abort(), right?

Reply:
yes, you are right. Here I just depend on the z_schedule inside k_sem_give to give way to other thread to abort it.  Thank you!

Reply:
updated, add a k_sleep here, waiting to be aborted.